OpenRA.Mods.Common/Lint/CheckUpgrades.cs

```diff
@@ -109,20 +109,6 @@ static IEnumerable<string> GetAllGrantedUpgrades(Action<string> emitError, Rules
 					}
 				}
 			}
-
-			// TODO: HACK because GainsExperience grants upgrades differently to most other sources.
-			var gainsExperience = rules.Actors.SelectMany(x => x.Value.TraitInfos<GainsExperienceInfo>()
-				.SelectMany(y => y.Upgrades.SelectMany(z => z.Value)));
-
-			foreach (var upgrade in gainsExperience)
-				yield return upgrade;
-
-			// TODO: HACK because Pluggable grants upgrades differently to most other sources.
-			var pluggable = rules.Actors.SelectMany(x => x.Value.TraitInfos<PluggableInfo>()
-				.SelectMany(y => y.Upgrades.SelectMany(z => z.Value)));
-
-			foreach (var upgrade in pluggable)
-				yield return upgrade;
```

Comment:
Can you also remove the similar hack in `GetAllUsedUpgrades` now, or does something else need to happen to kill that one off?